tools/chocolateyinstall.ps1

```diff
@@ -9,12 +9,32 @@ $url             = 'https://dl.influxdata.com/telegraf/releases/telegraf-1.4.0_w
 $url64           = 'https://dl.influxdata.com/telegraf/releases/telegraf-1.4.0_windows_amd64.zip'
 $fileLocation    = Join-Path $install_folder 'telegraf.exe'
 
+# Make sure the config directory exists
 If(!(Test-Path -Path $configDirectory)){
   New-Item -Path $configDirectory -ItemType Directory
 }
 
-If (Get-Service -Name "telegraf" -ErrorAction SilentlyContinue) {
+# If telegraf.exe exists, and the service is running, stop the service
+If (Test-Path -Path $fileLocation){
+  If (Get-Service -Name "telegraf" -ErrorAction SilentlyContinue) {
+    Stop-Service -Name "telegraf"
+    Start-Sleep -s 10
```

Comment:
Do we really need the sleep? Stop-Service will wait for the service to stop before it continues. If anything, 1 second should suffice.

Reply:
Yes - weird thing. Apparently doing a Stop-Service isn't asynchronous. The command exits immediately but the service actually isn't stopped. This was causing issues with the next steps of the install script and since the Stop-Service wasn't async, this was the only way I could "fix" it.